dascore/utils/misc.py

```
@@ -784,3 +784,98 @@ def maybe_mem_map(fid: IOBase, dtype="<u1") -> np.ndarray | np.memmap:
+    >>> deep_equality_check(dict1, dict2)
+    True
+    """
+
+    def _robust_equality_check(obj1, obj2):
+        """Robust equality check to also handle arrays."""
+        try:
+            equal = obj1 == obj2
+            # Handle numpy arrays and other array-like objects
+            if hasattr(equal, "all"):
+                return equal.all()
+            return equal
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)